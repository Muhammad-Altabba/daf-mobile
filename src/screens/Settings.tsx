/**
 * Serto Mobile App
 *
 */

import * as React from 'react'
import { useTranslation } from 'react-i18next'
import { Container, Text, Button, Constants, Screen } from '@kancha/kancha-ui'
import { NavigationScreenProps } from 'react-navigation'
import Config from 'react-native-config'

export default ({ navigation }: NavigationScreenProps) => {
  const { t, i18n } = useTranslation()
  return (
    <Screen scrollEnabled={true} safeArea={true}>
      <Container flex={1}>
        <Button
          type={Constants.BrandOptions.Primary}
          block={Constants.ButtonBlocks.Filled}
          buttonText={t('Logs')}
          onPress={() => navigation.navigate('Logs')}
          navButton
        />

        <Button
          type={Constants.BrandOptions.Primary}
          block={Constants.ButtonBlocks.Filled}
          buttonText={t('Signer')}
          onPress={() => navigation.navigate('Signer')}
          navButton
        />

        <Button
          type={Constants.BrandOptions.Primary}
          block={Constants.ButtonBlocks.Filled}
          buttonText={t('CodePush')}
          onPress={() => navigation.navigate('Codepush')}
          navButton
        />

        <Button
          type={Constants.BrandOptions.Primary}
          block={Constants.ButtonBlocks.Filled}
          buttonText={t('change')}
          onPress={() =>
            i18n.changeLanguage(i18n.language === 'es' ? 'en' : 'es')
          }
        />

        <Text type={Constants.TextTypes.H2}>{t('Configuration')}</Text>
        <Text type={Constants.TextTypes.Body}>{JSON.stringify(Config)}</Text>
      </Container>
    </Screen>
  )
}
