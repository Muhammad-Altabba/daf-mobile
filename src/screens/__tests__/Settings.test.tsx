import 'react-native'
import React from 'react'
import Settings from '../Settings'
import renderer from 'react-test-renderer'

jest.useFakeTimers()

it('renders correctly', () => {
  const tree = renderer.create(<Settings />).toJSON()
  expect(tree).toMatchSnapshot()
})
