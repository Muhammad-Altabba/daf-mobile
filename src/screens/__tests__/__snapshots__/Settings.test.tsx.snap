// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View>
    <Text
      style={
        Object {
          "color": "#FF0000",
          "fontSize": 20,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Settings
    </Text>
    <Text
      style={
        Object {
          "color": "#FF0000",
          "fontSize": 20,
          "margin": 10,
          "textAlign": "center",
        }
      }
    >
      Environment = 
      dev
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Log info
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Log warning
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Log error
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Press here to crash the app
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FF0000",
            "fontSize": 20,
            "margin": 10,
            "textAlign": "center",
          }
        }
      >
        Press here to trigger sample event
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Text>
        Check for updates
      </Text>
    </View>
    <Text>
      Update: 
      
    </Text>
  </View>
</RCTScrollView>
`;
